ta4j-core/src/main/java/org/ta4j/core/BaseTradingRecord.java

```diff
@@ -226,13 +228,31 @@ public boolean exit(int index, Num price, Num amount) {
 
     @Override
     public List<Position> getPositions() {
-        return positions;
+        return new LinkedList<>(sortedPositions.values());
```

Comment:
I can change it, no problem. (Although most criteria should iterate with for(Trade trade: trades) which is same speed for ArrayList and LinkedList). Personally, I would like to get rid of List's completely and would like to replace them with Collection's as the concrete type does not matter. Then we could just return sortedPositions.values() as it is

Reply:
Yes, change it to `ArrayList`. Don't return `Collection` because it does not explicitly express its ordering (which is important to _know_ for the calling methods, for example, `calculate(..)`); Returning a `List` is self-explanatory.